File: Sliding Window Tech/Slidingwind.py
def length_of_longest_substring(s):
    unique_chars = set()
    left = 0
    max_length = 0
    
    for right in range(len(s)):
        while s[right] in unique_chars:
            unique_chars.remove(s[left])
            left += 1
        
        unique_chars.add(s[right])
        max_length = max(max_length, right - left + 1)
    
    return max_length

File: Sliding Window Tech/Slidingwind.ts
function lengthOfLongestSubstring(s: string): number {
    const uniqueChars = new Set<string>();
    let left = 0;
    let maxLength = 0;
  
    for (let right = 0; right < s.length; right++) {
      while (uniqueChars.has(s[right])) {
        uniqueChars.delete(s[left]);
        left++;
      }
  
      uniqueChars.add(s[right]);
      maxLength = Math.max(maxLength, right - left + 1);
    }
  
    return maxLength;
  }